### PR TITLE
ObjectSearchCreate: searchFields or endpoint changing should clear results cache

### PR DESCRIPTION
**Pull request by rkuykendall**
**JIRA:** https://thatsmighty.atlassian.net/browse/MTY-962
Closes https://github.com/mighty-justice/fields-ant/issues/155

### CODE REVIEW

File: src/inputs/ObjectSearch.tsx [hunk 1]
```diff
@@ -72,6 +76,28 @@ class ObjectSearch extends Component<IObjectSearchProps> {
...
+
+  private get hasNewSearchFilters () {
+    return this.previousSearchFilters === toKey(this.fieldConfig.searchFilters || {});
+  }
+
+  private get isPristine () {
+    return (
+      !this.hasOptions
+      && !this.hasSearch
+      && !this.hasNewEndpoint
```
Comment (SHA-4) [inline src/inputs/ObjectSearch.tsx]: I believe this boolean logic is incorrect. You do want to search if there is a new endpoint or new search filters

File: test/types/objectSearchCreate.test.tsx [hunk 2]
```diff
@@ -217,4 +218,29 @@ describe('objectSearchCreate', () => {
 
     expect(onSave).toHaveBeenCalledWith({ law_firm: result });
   });
+
+  it('Properly caches and displays options', async () => {
+    const { field, searchTerm, result, props, endpoint } = getDefaults({})
+      , testerForm = await getTester(props) as any
```
Comment (SHA-4) [inline test/types/objectSearchCreate.test.tsx]: Instantiating tester to get props is a little roundabout. I would create mock props from a function; that way, when we write tests for just `ObjectSearch`, we'll have mocked props on the ready too

File: test/types/objectSearchCreate.test.tsx [hunk 3]
```diff
@@ -217,4 +218,29 @@ describe('objectSearchCreate', () => {
...
+    // First search
+    testerForm.endpoints['/legal-organizations/'] = { results: [result] };
+    await searchFor(tester, field, result, searchTerm);
+    expect(testerForm.getEndpoint.mock.calls.length).toBe(1);
+
+    // Re-focus but no new props
+    tester.instance.onFocus();
+    expect(testerForm.getEndpoint.mock.calls.length).toBe(1);
+
+    // Refocus but with new props
```
Comment (SHA-4) [inline test/types/objectSearchCreate.test.tsx]: Re-focus or refocus? Your comments conflict!

File: test/types/objectSearchCreate.test.tsx [hunk 4]
```diff
@@ -217,4 +218,29 @@ describe('objectSearchCreate', () => {
...
+    testerForm.endpoints['/legal-organizations-2/'] = { results: [result] };
```
Comment (SHA-4) [inline test/types/objectSearchCreate.test.tsx]: ${endpoint-2}?

File: test/types/objectSearch.test.tsx [hunk 5]
```diff
@@ -0,0 +1,68 @@
+import faker from 'faker';
+
+import { Tester } from '@mighty-justice/tester';
+
+import { fillInFieldConfig } from '../../src';
+import ObjectSearch from '../../src/inputs/ObjectSearch';
+
+function getDefaults (overrides?: any) {
```
Comment (SHA-4) [inline test/types/objectSearch.test.tsx]: Part of me wants this `getDefaults` to be a subset of / used in [this one](https://github.com/mighty-justice/fields-ant/blob/a8802f0f2325fd15105136fdfde6ea48b5f90498/test/types/objectSearchCreate.test.tsx#L8). What are your thoughts?

File: test/types/objectSearch.test.tsx [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+    field,
+    fieldConfig,
+    props,
+    type,
+
+    ...overrides,
+  };
+}
+
+async function searchFor (tester: any, field: string, result: any, searchTerm: string) {
```
Comment (SHA-4) [inline test/types/objectSearch.test.tsx]: This function is copied and pasted from [here](https://github.com/mighty-justice/fields-ant/blob/a8802f0f2325fd15105136fdfde6ea48b5f90498/test/types/objectSearchCreate.test.tsx#L48). I know these are specific to testing OSC components, but may as well import from this file or vice versa

File: test/types/objectSearch.test.tsx [hunk 7]
```diff
@@ -0,0 +1,68 @@
...
+
+describe('objectSearch', () => {
+  it('Properly caches and displays options', async () => {
+    const { field, searchTerm, result, props, endpoint } = getDefaults({})
+      , tester = (await new Tester(ObjectSearch, { props }).mount() as any);
+
+    expect(tester.getEndpoint.mock.calls.length).toBe(0);
+
+    // First search
+    tester.endpoints['/legal-organizations/'] = { results: [result] };
```
Comment (SHA-4) [inline test/types/objectSearch.test.tsx]: `searchFor` already includes this line

File: test/types/objectSearch.test.tsx (hunk 5 above)
Comment (rkuykendall) [reply test/types/objectSearch.test.tsx]: I actually prefer some copy-pasting in tests so that they work more independently.

**Pull request merged**